pkg/oci/config_validate.go

```diff
@@ -24,6 +24,28 @@ func validateAuthConfig(c *AuthConfig, fldPath *field.Path) field.ErrorList {
 	if c == nil {
 		return append(allErrs, field.Required(fldPath, ""))
 	}
+	if c.UseInstancePrincipals {
+		if c.Region != "" {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("region"), "cannot be used when useInstancePrincipals is enabled"))
+		}
+		if c.TenancyID != "" {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("tenancy"), "cannot be used when useInstancePrincipals is enabled"))
+		}
+		if c.CompartmentID != "" {
```

Comment:
Doesn't need validating as field depreciated (and the replacement field is actually still required for instance principles though it will be looked up dynamically via metadata if not present).